ring/model.go

```diff
@@ -192,8 +194,11 @@ func (d *Desc) mergeWithTime(mergeable memberlist.Mergeable, localCAS bool, now
 		return nil, nil
 	}
 
-	thisIngesterMap := buildNormalizedIngestersMap(d)
-	otherIngesterMap := buildNormalizedIngestersMap(other)
+	normalizeIngestersMap(d)
```

Comment:
Shouldn't the current value which we retrieved from `KV.store` already be normalized? If so, is it necessary to normalize `d` again, or could we assume that `d` is already normalized? I haven't checked how big of a performance difference this would be, might not be significant

Reply:
It is a decent win, I have numbers somewhere. It's just a bit hard to reason about so it's on the riskier end of the optimization spectrum.

Reply:
```
name                             old time/op    new time/op    delta
MemberlistReceiveWithRingDesc-6    4.12ms ±13%    3.58ms ± 8%  -13.11%  (p=0.000 n=10+9)

name                             old alloc/op   new alloc/op   delta
MemberlistReceiveWithRingDesc-6    51.6kB ±20%    32.0kB ±15%  -37.95%  (p=0.000 n=10+10)

name                             old allocs/op  new allocs/op  delta
MemberlistReceiveWithRingDesc-6       681 ± 2%        75 ± 7%  -89.03%  (p=0.000 n=10+10)
```